File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.ts
function minDeletionSize(strs: string[]): number {

    var deletedColumns = 0; 
        var szCol = strs[0].length;
        var szRow = strs.length;
        var i  = 0; 
        while(i< szCol)
        {
            var last = strs[0][i];
            var j = 1; 
            while(j< szRow)
            {
                if(strs[j][i] < last)
                {
                    deletedColumns++;
                    break;
                }
                last = strs[j][i];
                j++;
            }
            i++;
        }
        return deletedColumns;
    
};

File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {

        int deletedColumns = 0; 
        int szCol = (int)strs[0].size();
        int szRow = (int)strs.size();
        int i  = 0; 
        while(i< szCol)
        {
            char last = strs[0][i];
            int j = 1; 
            while(j< szRow)
            {
                if(strs[j][i] < last)
                {
                    deletedColumns++;
                    break;
                }
                last = strs[j][i];
                j++;
            }
            i++;
        }
        return deletedColumns;
    }
};